tests/iterators.rs

```diff
@@ -235,6 +252,31 @@ fn outer_iter_mut() {
 }
 
 #[test]
+fn axis_iter_mut() {
+    let a = Array::from_iter(0..12);
+    let a = a.reshape((2, 3, 2));
+    // [[[0, 1],
+    //   [2, 3],
+    //   [4, 5]],
+    //  [[6, 7],
+    //   [8, 9],
+    //    ...
+    let mut a = a.to_owned();
+
+    for mut subview in a.axis_iter_mut(1) {
```

Comment:
`mut subview`; it's sad quirks like this because we have no user-defined DSTs (views should be unsized like slices I think).